cpp/tests/column/column_view_device_span_test.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <memory>
+#include <type_traits>
+
+// fixed_width, dict, string, list, struct
+template <typename T, std::enable_if_t<cudf::is_fixed_width<T>()>* = nullptr>
+std::unique_ptr<cudf::column> example_column()
+{
+  auto begin = thrust::make_counting_iterator(1);
+  auto end   = thrust::make_counting_iterator(16);
+  return cudf::test::fixed_width_column_wrapper<T>(begin, end).release();
+}
+
+template <typename T>
+struct ColumnViewDeviceSpanTests : public cudf::test::BaseFixture {
+};
+
+using DeviceSpanTypes =
+  cudf::test::Types<int32_t>;  // cudf::test::FixedWidthTypesWithoutFixedPoint;
+TYPED_TEST_SUITE(ColumnViewDeviceSpanTests, DeviceSpanTypes);
+
+TYPED_TEST(ColumnViewDeviceSpanTests, conversion_round_trip)
+{
+  auto col                   = example_column<TypeParam>();
+  auto col_view              = cudf::column_view{*col};
+  auto device_span_from_view = static_cast<cudf::device_span<const TypeParam>>(col_view);
+  auto col_view_from_span    = cudf::column_view(device_span_from_view);
```

Comment:
@jrhemstad Can you give a bit more context for why you think implicit conversions are more appropriate than explicit conversions? @vuule recommended it should be explicit.

Reply:
Because the precedent for things like `string_view`, `span`, and `std::ranges::views` is implicit conversion. 

wrt to launching a kernel in `column_view` -> `span`, I suggest only checking `nullable` instead of `null_count`, as otherwise there is no mechanism to inject a stream. You could add an explicit `column_view::to_span(cudaStream_t)` that would instead check `null_count()` and pass the stream when `null_count` is made to expose one.

Reply:
Thank you for the info! That all makes sense. I considered checking `nullable` but was worried that we might want to support columns that have a fully-valid null mask (instead of a mask that is `nullptr`). However, I agree that it's not worth handling the potential kernel launch / stream parameter. It is now checking `nullable` instead. That covers everything in this discussion, so I'll mark this thread as resolved.